### ISSUE TITLE
Request: Add option to TextEditor.enrichHTML() to not roll inline rolls but still enrich them

### ISSUE THREAD

**Issue by mkahvi** [foundryvtt/foundryvtt#10557]
### User Experience

For example if I include `[[3d6]]` in an item description, I'd like to have an option to not roll that and just display the formula kind of like how `[[/r 3d6]]` does.

This is mostly for displaying item descriptions in the item sheet or embedded into actor sheet, so they show the roll formula, but don't roll it every time the sheet is opened or updated in any manner.

I personally only need the rolling to happen only once the inline roll hits chat log, and it's much more informative to user before that to see the formula instead, but I don't want to require clicking like the `/r` variant does to accomplish it.

This is not doable by passing `rolls:false` option to the enricher, since that leaves them entirely untouched, though that does make it possible to implement this locally.

**Comment by JPMeehan** [comment]
This would also be useful for displaying Roll Table results in an "un-finalized" form.

**Comment by mclemente** [comment]
You can remedy this by doing `[[3d6]]{3d6}`, which will still display a roll result, but at least the formula will be kept.

**Comment by aaclayton** [comment]
I don't really understand this request. Why would you not be using `[[/r {formula}]]`? The use case here is not clear to me.

**Comment by mkahvi** [comment]
I want the inline roll to be automatically rolled when the item description is printed to chat, not require separate click and separate card with that.

**Comment by aaclayton** [comment]
In that case, in the HTML markup it should be a non-immediate roll and the roll should be handled as part of the "play card to chat" workflow. This isn't the right way to think about that problem. Closing as wontfix.

**Comment by mkahvi** [comment]
So the solution still remains to use rolls:false option and have custom attempt to render inline rolls? Meh.

**Comment by dmrickey** [comment]
> I don't really understand this request. Why would you not be using `[[/r {formula}]]`? The use case here is not clear to me.

Viewing "This ability can be used in `1d4` rounds" when viewing the description within the Item itself is very different from reading "This ability can be used in `3` rounds" when the Item's chat card is printed because the item was just used

**Comment by aaclayton** [comment]
> Viewing "This ability can be used in `1d4` rounds" when viewing the description within the Item itself is very different from reading "This ability can be used in `3` rounds" when the Item's chat card is printed because the item was just used

That is **exactly** what the `[[/r 1d4]]` syntax exists to do. This already exists.

**Comment by mkahvi** [comment]
The `[[/r 1d4]]` syntax requires additional interaction and another card disconnected from the first. It's far from the same thing.

The main desire here is that I can use `[[1d4]]` in item description and that it does not roll __when viewing the item sheet__.

**Comment by dmrickey** [comment]
mkahvi can correct me if I'm wrong, but here's the difference in setup.

![image](https://github.com/foundryvtt/foundryvtt/assets/3664822/5d349036-30f0-4cf5-b43a-30f5ff76eb1e)
![image](https://github.com/foundryvtt/foundryvtt/assets/3664822/6d5aad99-f5ac-43aa-8522-384cf871b6df)

I believe the goal is for the compendium item/description to actually display `1d4` per the `[[/r 1d4]]` example, but to roll the value in the chat card per the `[[1d4]]` example without requiring user interaction. I'm sure there are exceptions, but it seems like the majority of the time when you're reading an item description or a compendium entry, you need to know the randomness of it, but by the time it hits the chat card you need the outcome of the roll.

So the desire is to show `1d4` in the description, and `the result of 1d4` in the chat card without having to write a custom solution

**Comment by aaclayton** [comment]
Chat cards are owned by the game system or module that created them. Core does not create these chat cards. You could request the game system to implement this feature to automatically evaluate any deferred rolls that are part of the chat card HTML. Not sure whether that request would be prioritized but it would be a more appropriate place than here.

**Comment by JPMeehan** [comment]
I think there's a specific use case with Rollable Tables, where if you're viewing the results  _as a table_ you wouldn't want the evaluated version, while if you're outputting a specific result you'd want any rolls to be evaluated

**Comment by mkahvi** [comment]
> You could request the game system to implement this feature to automatically evaluate any deferred rolls that are part of the chat card HTML. 

That would then deny use of deferred rolls from users who want to use those.

**Comment by mkahvi** [comment]
Anyone else needing this can check PF1 for an example implementation that isn't pretty: <https://gitlab.com/foundryvtt_pathfinder1e/foundryvtt-pathfinder1/-/blob/e62b9575b13ff4c51e15b45b0ff34cc0ae77ef2e/module/utils/lib.mjs#L587-698>

Needed to change it and re-implement way more functions for v13 to get it working.
